tests/test_manifest.py

```diff
@@ -203,6 +204,92 @@ def session_func():
     assert len(manifest) == 2
 
 
+def test_extra_python_no_python():
+    cfg = mock.sentinel.CONFIG
+    cfg.force_venv_backend = None
+    cfg.default_venv_backend = None
+    cfg.extra_pythons = ["3.8"]
+
+    manifest = Manifest({}, cfg)
+
+    def session_func():
+        pass
+
+    func = Func(session_func)
+    for session in manifest.make_session("my_session", func):
+        manifest.add_session(session)
+
+    assert len(manifest._all_sessions) == 1
+    assert len(manifest) == 1
+    session_friendly_names = [s.friendly_name for s in manifest._all_sessions]
+    assert session_friendly_names == ["my_session"]
+    session_pythons = [s.func.python for s in manifest._all_sessions]
+    assert session_pythons == ["3.8"]
+
+def test_extra_python_no_python_multiples():
+    cfg = mock.sentinel.CONFIG
+    cfg.force_venv_backend = None
+    cfg.default_venv_backend = None
+    cfg.extra_pythons = ["3.8", "3.9"]
+
+    manifest = Manifest({}, cfg)
+
+    def session_func():
+        pass
+
+    func = Func(session_func)
+    for session in manifest.make_session("my_session", func):
+        manifest.add_session(session)
+
+    assert len(manifest._all_sessions) == 2
+    assert len(manifest) == 2
+    session_friendly_names = [s.friendly_name for s in manifest._all_sessions]
+    assert session_friendly_names == ["my_session-3.8", "my_session-3.9"]
+    session_pythons = [s.func.python for s in manifest._all_sessions]
+    assert session_pythons == ["3.8", "3.9"]
+
+def test_extra_python_single_python():
+    cfg = mock.sentinel.CONFIG
+    cfg.force_venv_backend = None
+    cfg.default_venv_backend = None
+    cfg.extra_pythons = ["3.8"]
+
+    manifest = Manifest({}, cfg)
+
+    def session_func():
+        pass
+
+    func = Func(session_func, python="3.5")
+    for session in manifest.make_session("my_session", func):
+        manifest.add_session(session)
+
+    assert len(manifest._all_sessions) == 2
+    assert len(manifest) == 2
+    session_pythons = [s.func.python for s in manifest._all_sessions]
+    assert session_pythons == ["3.5", "3.8"]
+
+
+def test_extra_python_multiple_python():
+    cfg = mock.sentinel.CONFIG
+    cfg.force_venv_backend = None
+    cfg.default_venv_backend = None
+    cfg.extra_pythons = ["3.8", "3.4"]
+
+    manifest = Manifest({}, cfg)
+
+    def session_func():
+        pass
+
+    func = Func(session_func, python=["3.5", "3.9"])
+    for session in manifest.make_session("my_session", func):
+        manifest.add_session(session)
+
+    assert len(manifest._all_sessions) == 4
+    assert len(manifest) == 4
+    session_pythons = [s.func.python for s in manifest._all_sessions]
+    assert session_pythons == ["3.5", "3.9", "3.8", "3.4"]
+
+
```

Comment:
Adapt tests to change in behavior when python is None or False. 

Replace test cases by a single parametrized test, and add a few additional cases to the parametrization.

```suggestion
@pytest.mark.parametrize(
    "python,extra_pythons,expected",
    [
        (None, [], [None]),
        (None, ["3.8"], [None]),
        (None, ["3.8", "3.9"], [None]),
        (False, [], [False]),
        (False, ["3.8"], [False]),
        (False, ["3.8", "3.9"], [False]),
        ("3.5", [], ["3.5"]),
        ("3.5", ["3.8"], ["3.5", "3.8"]),
        ("3.5", ["3.8", "3.9"], ["3.5", "3.8", "3.9"]),
        (["3.5", "3.9"], [], ["3.5", "3.9"]),
        (["3.5", "3.9"], ["3.8"], ["3.5", "3.9", "3.8"]),
        (["3.5", "3.9"], ["3.8", "3.4"], ["3.5", "3.9", "3.8", "3.4"]),
    ],
)
def test_extra_pythons(python, extra_pythons, expected):
    cfg = mock.sentinel.CONFIG
    cfg.force_venv_backend = None
    cfg.default_venv_backend = None
    cfg.extra_pythons = extra_pythons

    manifest = Manifest({}, cfg)

    def session_func():
        pass

    func = Func(session_func, python=python)
    for session in manifest.make_session("my_session", func):
        manifest.add_session(session)

    assert expected == [session.func.python for session in manifest._all_sessions]


```